src/editors/colorpicker.js

```diff
@@ -0,0 +1,130 @@
+  window.addEventListener('mousedown', mouseDownListener) // listen the mouseevent of window
+}
+
+function mouseDownListener (e) {
+  var colorPicker = currentColorPicker
+  // return the colorpciker root dom element or false
+  var isColorPicker = colorPicker && (function (elm) {
+    while (elm) {
+      if ((elm.className || '').indexOf('cp-app') !== -1) return elm
+      elm = elm.parentNode
+    }
+    return false
+  })(e.target)
+  if (isColorPicker) {
+    if (e.target === colorPicker.nodes.exit) {
+      colorPicker.node.colorPicker.style.display = 'none'
+      document.activeElement.blur()
+    }
+  } else if (currInstance && e.target !== currInstance.input) {
+    colorPicker.nodes.colorPicker.style.display = 'none'
+  }
+}
+export var ColorEditor = StringEditor.extend({
+
+  build: function () {
+    this._super()
+    if (!this.input) return
+
+    if (window.ColorPicker) {
+      this.input.setAttribute('type', 'text')
+      const options = (this.options || {}).ColorPicker || {}
+      this.isAlpha = false
+      this.format = 'hex'
+      if (options.format) {
+        this.format = options.format.toLowerCase()
+        if (options.format.indexOf('a') !== -1) {
+          this.isAlpha = true
+        }
+        this.format = this.format.replace(/a/g, '')
+      }
+      if (!this.created) {
+        if (!currentColorPicker) {
+          createColorPicker()
+        }
+        this.input.style.backgroundColor = this.value
+        this.created = true
+        this.doEventListeners(false) // listen focus event
+      }
+    }
+  },
+  doEventListeners: function (off) {
+    var onOff = off ? 'removeEventListener' : 'addEventListener'
+    var self = this
+    // do when colorpicker input is focusing
+    var focusListener = function (e) {
+      var input = this
+      var position = window.ColorPicker.getOrigin(input)
+      var options = currentColorPicker.color.options
+      var colorPickerUI = currentColorPicker.nodes.colorPicker
+      var appendTo = document.body
+      var isStatic = /static/.test(window.getComputedStyle(appendTo).position)
+      var atrect = isStatic ? {left: 0, top: 0} : appendTo.getBoundingClientRect()
+      var waitTimer = 0
+      options.color = self.value // brings color to default on reset
+      currInstance = self
+      colorPickerUI.style.cssText =
+              'position: absolute;' + (!currentColorPicker.cssIsReady ? 'display: none;' : '') +
+              'left:' + (position.left + options.margin.left - atrect.left) + 'px;' +
+              'top:' + (position.top + input.offsetHeight + options.margin.top - atrect.top) + 'px;'
+      currentColorPicker.setColor(self.value || '#000000', undefined, undefined, true)
+      waitTimer = setInterval(function () { // compensating late style on onload in colorPicker
+        if (currentColorPicker.cssIsReady) {
+          waitTimer = clearInterval(waitTimer)
+          colorPickerUI.style.display = 'block'
+        }
+      }, 10)
```

Comment:
@hybtalented There's a function in `src/editors/string.js`, named `afterInputReady` Anything that should happen when DOM element is ready (Like initializing/applying plugins), should be placed there. 
Take a look at `src/editors/jodit.js`, which is extremely simple and also shows how to access the global/schema options using the new 2.x method.

Reply:
Ok, I will try the vanilla-picker instead.

Reply:
@pmk65, the colorpicker is  use the https://github.com/Sphinxxxx/vanilla-picker library now